sdk/core/azure-core/src/main/java/com/azure/core/util/Option.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import java.util.NoSuchElementException;
+
+/**
+ * The Option type to describe tri-state. Every Option instance is in one of
+ * the three states: a state representing a non-null-value, null-value, or no-value.
+ *
+ * @param <T> The value type.
+ */
+public final class Option<T> {
+    private static final Option<?> UNINITIALIZED = new Option<>();
+    private static final Option<?> EMPTY = new Option<>(null);
+    private final boolean isInitialized;
+    private final T value;
+
+    /**
+     * Returns an {@link Option} with the specified null-value or non-null-value.
+     *
+     * @param <T> The value type.
+     * @param value the value.
+     * @return an {@link Option} with the value present.
+     */
+    public static <T> Option<T> of(T value) {
+        return value == null ? empty() : new Option<>(value);
+    }
+
+    /**
+     * Returns an {@link Option} with null-value.
+     * <p>
+     * {@code Option.empty()} is a syntactic sugar for {@code Option.of(null)}.
+     * </p>
+     * @param <T> The value type.
+     * @return an {@link Option} with a null-value.
+     */
+    public static <T> Option<T> empty() {
+        @SuppressWarnings("unchecked")
+        Option<T> empty = (Option<T>) EMPTY;
+        return empty;
+    }
+
+    /**
+     * Returns an {@link Option} instance with no-value.
+     *
+     * @param <T> Type of the non-existent value.
+     * @return An Option type with no-value.
+     */
+    public static <T> Option<T> uninitialized() {
```

Comment:
Should we allow uninitialized `Option` to be initialized later after creation by calling `setValue(T value)`?

Reply:
pls refer [comment](https://github.com/Azure/azure-sdk-for-java/pull/15565#discussion_r506759874)